company/serializers.py

```diff
@@ -20,33 +23,50 @@ class Meta:
 
 
+class ImageSerializer(serializers.ModelSerializer):
+    """Special Serializer to customize validators=[validate_image_size]"""
+    image = serializers.ImageField(use_url=True, allow_null=True, validators=[validate_image_size])
+
+    class Meta:
+        model = Image
+        fields = ['image']
+
+
 class CreateCompanySerializer(serializers.ModelSerializer):
     password = serializers.CharField(write_only=True, required=True)
     owner = serializers.ReadOnlyField(source='owner.email')
     legal_address = AddressSerializer()
     actual_address = AddressSerializer()
+    logo = ImageSerializer()
 
     class Meta:
         model = Company
         fields = ['id', 'owner', 'legal_name', 'logo', 'legal_address', 'actual_address', 'code_USREOU', 'phone',
                   'email', 'password']
 
     def create(self, validated_data):
-        legal_address = validated_data.pop('legal_address')
-        actual_address = validated_data.pop('actual_address')
-        company = Company.objects.create(
-            legal_address=Address.objects.create(**legal_address),
-            actual_address=Address.objects.create(**actual_address),
-            **validated_data,
-        )
+        legal_address_data = validated_data.pop('legal_address')
+        actual_address_data = validated_data.pop('actual_address')
+        logo_data = validated_data.pop('logo')
+        with transaction.atomic():  # atomic should stop other requests to db until we make the changes
+            """ Works only with custom ImageSerializer.
+            Does not work directly through the serializers.ImageField."""
+            company = Company.objects.create(
+                legal_address=Address.objects.create(**legal_address_data),
+                actual_address=Address.objects.create(**actual_address_data),
+                logo=Image.objects.create(**logo_data),
+                **validated_data,
+            )
         return company
 
     def update(self, instance, validated_data):
-        legal_address = validated_data.pop('legal_address')
-        actual_address = validated_data.pop('actual_address')
+        legal_address_data = validated_data.pop('legal_address')
+        actual_address_data = validated_data.pop('actual_address')
+        logo_data = validated_data.pop('logo')
         # rewrite the address separately by using 'Serializer'
-        instance.legal_address = AddressSerializer().update(instance.legal_address, legal_address)
-        instance.actual_address = AddressSerializer().update(instance.actual_address, actual_address)
+        instance.legal_address = AddressSerializer().update(instance.legal_address, legal_address_data)
+        instance.actual_address = AddressSerializer().update(instance.actual_address, actual_address_data)
+        instance.logo = ImageSerializer().update(instance.logo, logo_data)
         return super().update(instance, validated_data)  # rewrite company by default method - 'update'
```

Comment:
is this comment still necessary? is it valid?

Reply:
done